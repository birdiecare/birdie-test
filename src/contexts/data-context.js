import React, { createContext, useContext } from 'react';
import { oneOfType, arrayOf, node } from 'prop-types';
import axios from 'axios';

import { BASE_APPLICATION } from 'gatsby-env-variables';

export const DataContext = createContext();

export const useDataContext = () => useContext(DataContext) || {};

export const DataProvider = ({ children }) => (
          <DataContext.Provider
            value={{
              getMood: async () => {
                const { data } = await axios({
                  method: 'get',
                  url: `${BASE_APPLICATION}/get-mood`,
                  headers: { 'Content-Type': 'application/json' },
                });
                const { mood: { mood } } = data;
                return mood;
              },
            }}>
            {children}
          </DataContext.Provider>
);

DataProvider.propTypes = {
  children: oneOfType([arrayOf(node), node]).isRequired,
};
