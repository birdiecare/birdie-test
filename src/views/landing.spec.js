import React from 'react';
import { shallow } from 'enzyme';

import { useDataContext } from 'contexts/data-context';
import { Button } from '@material-ui/core';

import LandingView from './landing';

jest.mock('contexts/data-context');

const render = ({ mood = '' } = {}) => shallow(<LandingView {...{ mood }} />);

describe('LandingView renders', () => {
  const getMood = () => {};

  beforeEach(() => {
    jest.clearAllMocks();
    useDataContext.mockReturnValue({ getMood });
  });

  test('default render is as expected', () => expect(render()).toMatchSnapshot());

  ['sad', 'tired', 'confused', 'hungry'].forEach((mood) => {
    test(`mood ${mood} is rendered as expected`, () => expect(render({ mood })).toMatchSnapshot());
  });
});

describe('LandingView icon changes when button is pressed', () => {
  const getMood = jest.fn();
  let landingView;
  beforeAll(() => {
    jest.clearAllMocks();
    getMood.mockReturnValue('sad');
    useDataContext.mockReturnValue({ getMood });
    landingView = render({ loading: false });
  });

  test('button is rendered', () => {
    const button = landingView.find(Button);
    expect(button.exists()).toEqual(true);
  });

  test('button click changes mood', () => {
    const button = landingView.find(Button);
    button.simulate('click');
    expect(getMood).toHaveBeenCalled();
  });
});
