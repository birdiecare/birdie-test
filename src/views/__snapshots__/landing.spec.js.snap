// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingView renders default render is as expected 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  direction="column"
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <Mood
      mood=""
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Button))
      aria-label="update mood"
      color="secondary"
      endIcon={<Memo(UpdateIcon) />}
      onClick={[Function]}
      size="large"
      variant="outlined"
    >
      Update mood
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`LandingView renders mood confused is rendered as expected 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  direction="column"
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <Mood
      mood="confused"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Button))
      aria-label="update mood"
      color="secondary"
      endIcon={<Memo(UpdateIcon) />}
      onClick={[Function]}
      size="large"
      variant="outlined"
    >
      Update mood
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`LandingView renders mood hungry is rendered as expected 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  direction="column"
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <Mood
      mood="hungry"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Button))
      aria-label="update mood"
      color="secondary"
      endIcon={<Memo(UpdateIcon) />}
      onClick={[Function]}
      size="large"
      variant="outlined"
    >
      Update mood
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`LandingView renders mood sad is rendered as expected 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  direction="column"
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <Mood
      mood="sad"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Button))
      aria-label="update mood"
      color="secondary"
      endIcon={<Memo(UpdateIcon) />}
      onClick={[Function]}
      size="large"
      variant="outlined"
    >
      Update mood
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`LandingView renders mood tired is rendered as expected 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  direction="column"
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <Mood
      mood="tired"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Button))
      aria-label="update mood"
      color="secondary"
      endIcon={<Memo(UpdateIcon) />}
      onClick={[Function]}
      size="large"
      variant="outlined"
    >
      Update mood
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
